metadata-ingestion/src/datahub/ingestion/source/mlflow.py

```diff
@@ -262,6 +264,92 @@ def _convert_run_result_type(
                 type="SKIPPED", nativeResultType=self.platform
             )
 
+    def _get_dataset_schema(self, schema: str) -> Optional[List[Tuple[str, str]]]:
+        try:
+            schema_dict = json.loads(schema)
+        except json.JSONDecodeError:
+            print("Failed to parse schema JSON")
+            return None
+
+        if "mlflow_colspec" in schema_dict:
+            try:
+                return [
+                    (field["name"], field["type"])
+                    for field in schema_dict["mlflow_colspec"]
+                ]
+            except (KeyError, TypeError):
+                return None
+        # If the schema is not formatted, return None
+        return None
+
+    def _get_dataset_platform_from_source_type(self, source_type):
+        # manually map mlflow platform to datahub platform
+        if source_type == "gs":
+            return "gcs"
+        return source_type
+
+    def _get_dataset_input_workunits(self, run: Run) -> Iterable[MetadataWorkUnit]:
+        run_urn = DataProcessInstance(
+            id=run.info.run_id,
+            orchestrator=self.platform,
+        ).urn
+        dataset_inputs = run.inputs.dataset_inputs
+        dataset_reference_urns = []
+        for dataset_input in dataset_inputs:
+            source_type = dataset_input.dataset.source_type
+            dataset_tags = {k[1]: v[1] for k, v in dataset_input.tags}
+            dataset = dataset_input.dataset
+            platform = self._get_dataset_platform_from_source_type(source_type)
+            custom_properties = dataset_tags
+            formatted_schema = self._get_dataset_schema(dataset.schema)
+            # If the schema is not formatted, pass the schema as a custom property
+            if formatted_schema is None:
+                custom_properties["schema"] = dataset.schema
+            # If the dataset is local or code, we create a local dataset reference
+            if source_type in ("local", "code"):
+                local_dataset_reference = Dataset(
+                    platform=platform,
+                    name=dataset.name,
+                    schema=formatted_schema,
+                    custom_properties=custom_properties,
+                )
+                yield from local_dataset_reference.as_workunits()
+                dataset_reference_urns.append(str(local_dataset_reference.urn))
+            # Otherwise, we create a hosted dataset reference and a hosted dataset
+            else:
+                hosted_dataset = Dataset(
```

Comment:
imo we should not be generating dataset entities for non mlflow platforms

here we should just do `hosted_dataset_urn = DatasetUrn...`. if that urn exists, lineage will show up by default. if it doesn't exist, they'll need to go into the UI and click "show hidden edges" to make them show up